packages/fether-react/src/Send/TxForm/TxForm.js

```diff
@@ -39,21 +40,22 @@ class Send extends Component {
   state = {
     maxSelected: false
   };
-  handleSubmit = values => {
-    const { accountAddress, history, sendStore, token } = this.props;
-
-    sendStore.setTx(values);
-    history.push(`/send/${token.address}/from/${accountAddress}/signer`);
-  };
 
   decorator = createDecorator({
     field: /to|amount/, // when the value of these fields change...
     updates: {
       // ...set field "gas"
-      gas: (value, allValues) => {
+      gas: async (value, allValues) => {
         const { parityStore, token } = this.props;
+
         if (this.preValidate(allValues) === true) {
-          return estimateGas(allValues, token, parityStore.api);
+          const newEstimatedTxFee = await estimateGas(
```

Comment:
Hmm, I'm not sure what happens when `estimateGas` throws an error, it depends on how final-form-calculate is implemented. To be sure, let's add a try/catch here?

(^ not introduced by this PR, this potential bug was here before, I just saw it).